woocommerce/google-listings-and-ads
# Remove IdTrait and retrieve from Options

### Changes proposed in this Pull Request:

The Options class is already managing the Merchant and Ads account ID's (also forcing them to be a PositiveInteger). Most of the classes accessing the ID's already have access to the Options instance. So this PR always retrieves the account ID's directly from Options so we never have to deal with any stale copies of the account ID. Since we allow overriding the ID through a query variable (currently only used for the ConnectionTest page), we need to implement some helper functions. This can be refactored once we no longer rely on the ConnectionTest page.

These changes allow us to do the following:
- Remove AdsIdTrait
- Remove any calls to `set_id`, which was intended to update a stale copy
- Remove the need to pass the account ID to each class in the GoogleServiceProvider

Closes #391

### Detailed test instructions:

1. Check requests on the ConnectionTest page and confirm that they all still work and return the status we expect
2. Try to send a proxied request to Merchant Center or Ads with a different account ID to confirm it allows overrides

### Changelog Note:
- Remove IdTrait and retrieve account Ids directly from Options.
